src/backend/api/meals.js

```diff
@@ -2,13 +2,125 @@ const express = require("express");
 const router = express.Router();
 const knex = require("../database");
 
-router.get("/", async (request, response) => {
+router.get("/", async (req, res) => {
   try {
-    // knex syntax for selecting things. Look up the documentation for knex for further info
-    const titles = await knex("meals").select("title");
-    response.json(titles);
+    const meals = knex("homework3.Meal");
+
+    const {
+      maxPrice,
+      availableReservations,
+      title,
+      dateAfter,
+      dateBefore,
+      limit,
+      sortKey,
+      sortDir
+    } = req.query;
+
+    if (maxPrice !== undefined) {
```

Comment:
In JavaScript, usually when checking if a value is defined or undefined/null, you can just parse the value as a boolean, i.e. `if (maxPrice) {...}`. This is due to a concept called Truthy / Falsy in JavaScript: https://developer.mozilla.org/en-US/docs/Glossary/Falsy

The point here is that often we are interested to check if a value is null, or undefined, or empty string etc, and esp. with null and undefined, it's always safer to check for both. That's where you could use the Falsy concept. 

Notice that 0 is also a "Falsy" value (see the link above). So for example if you have a user field where 0 is an actual valid value, you need to include that as well in the check, for example `if (maxPrice || maxPrice === 0) {...}`.

Reply:
Super, is a nice alternative, I will keep it in mind :)